File: app/src/main/java/com/autohub/skln/fragment/FragmentClassRequests.java
package com.autohub.skln.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.autohub.skln.R;
import com.autohub.skln.databinding.FragmentClassRequestsBinding;
import com.autohub.skln.models.User;
import com.autohub.skln.utills.AppConstants;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.DocumentSnapshot;

import java.util.ArrayList;
import java.util.List;

public class FragmentClassRequests extends BaseFragment {
    private String mType = "student";
    private FragmentClassRequestsBinding mBinding;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mType = getArguments().getString(AppConstants.KEY_TYPE, "student");
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_class_requests, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mBinding = FragmentClassRequestsBinding.bind(view);
        fetchUser();

    }

    private void fetchUser() {
        String root = getString(R.string.db_root_tutors);
        if (mType.equalsIgnoreCase("student")) {
            root = getString(R.string.db_root_students);
        }
        getFirebaseStore().collection(root).document(getFirebaseAuth().getCurrentUser().getUid()).get()
                .addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                    @Override
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                        User user = documentSnapshot.toObject(User.class);
                        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
                        adapter.addData(getFragmentClassRequests("Latest", user), "Latest");
                        adapter.addData(getFragmentClassRequests("All", user), "All");
                        mBinding.tabs.setupWithViewPager(mBinding.viewpager);
                        mBinding.viewpager.setAdapter(adapter);

                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        showSnackError(e.getMessage());
                    }
                });
    }

    private FragmentRequests getFragmentClassRequests(String type, User user) {
        FragmentRequests latestRequests = new FragmentRequests();
        Bundle bundle = new Bundle();
        bundle.putString(AppConstants.KEY_TYPE, type);
        bundle.putParcelable(AppConstants.KEY_DATA, user);
//        String root = "Student";
//        if (mType.equalsIgnoreCase("student")) {
//            root = "Tutor";
//        }
        bundle.putString("_user_type", mType);
        latestRequests.setArguments(bundle);
        return latestRequests;
    }

    private static class ViewPagerAdapter extends FragmentPagerAdapter {
        private final List<Info> mData = new ArrayList<>();

        ViewPagerAdapter(FragmentManager manager) {
            super(manager);
        }

        @NonNull
        @Override
        public Fragment getItem(int position) {
            return mData.get(position).fragment;
        }

        @Override
        public int getCount() {
            return mData.size();
        }

        void addData(Fragment fragment, String title) {
            mData.add(new Info(fragment, title));
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return mData.get(position).title;
        }

        @Override
        public long getItemId(int position) {
            return System.currentTimeMillis();
        }

        class Info {
            final Fragment fragment;
            final String title;

            Info(Fragment fragment, String title) {
                this.fragment = fragment;
                this.title = title;
            }
        }
    }
}

File: studentmodule/src/main/java/com/netzwelt/studentmodule/fragments/FragmentClassRequests.kt
package com.netzwelt.studentmodule.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.autohub.skln.fragment.BaseFragment
import com.autohub.skln.models.User
import com.autohub.skln.utills.AppConstants
import com.netzwelt.studentmodule.R
import com.netzwelt.studentmodule.databinding.FragmentClassRequestsBinding

import java.util.ArrayList

class FragmentClassRequests : BaseFragment() {
    private var mType = "student"
    private var mBinding: FragmentClassRequestsBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
       /* if (arguments != null) {
            mType = arguments!!.getString(AppConstants.KEY_TYPE, "student")
        }*/
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_class_requests, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mBinding = FragmentClassRequestsBinding.bind(view)
        fetchUser()

    }

    private fun fetchUser() {
        var root = getString(R.string.db_root_tutors)
        if (mType.equals("student", ignoreCase = true)) {
            root = getString(R.string.db_root_students)
        }
        firebaseStore.collection(root).document(firebaseAuth.currentUser!!.uid).get()
                .addOnSuccessListener { documentSnapshot ->
                    val user = documentSnapshot.toObject(User::class.java)
                    val adapter = ViewPagerAdapter(childFragmentManager)
                    adapter.addData(getFragmentClassRequests("Latest", user), "Latest")
                    adapter.addData(getFragmentClassRequests("All", user), "All")
                    mBinding!!.tabs.setupWithViewPager(mBinding!!.viewpager)
                    mBinding!!.viewpager.adapter = adapter
                }
                .addOnFailureListener { e -> showSnackError(e.message) }
    }

    private fun getFragmentClassRequests(type: String, user: User?): FragmentRequests {
        val latestRequests = FragmentRequests()
        val bundle = Bundle()
        bundle.putString(AppConstants.KEY_TYPE, type)
        bundle.putParcelable(AppConstants.KEY_DATA, user)
        var root = "Student"
        if (mType.equals("student", ignoreCase = true)) {
            root = "Tutor"
        }
        bundle.putString("_user_type", mType)
        latestRequests.arguments = bundle
        return latestRequests
    }

    private class ViewPagerAdapter internal constructor(manager: FragmentManager) : FragmentPagerAdapter(manager) {
        private val mData = ArrayList<Info>()

        override fun getItem(position: Int): Fragment {
            return mData[position].fragment
        }

        override fun getCount(): Int {
            return mData.size
        }

        internal fun addData(fragment: Fragment, title: String) {
            mData.add(Info(fragment, title))
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return mData[position].title
        }

        override fun getItemId(position: Int): Long {
            return System.currentTimeMillis()
        }

        internal inner class Info(val fragment: Fragment, val title: String)
    }
}